db/migrate/20240729080631_electric_vehicle_profiles.rb

```diff
@@ -0,0 +1,174 @@
+
+  OLD_SHARE_KEYS = NEW_SHARE_MAPPING.keys.map(&:to_s)
+  NEW_SHARE_KEYS = NEW_SHARE_MAPPING.values
+
+  # Executes the migration for all scenarios.
+  def up
+    migrate_scenarios do |scenario|
+      # Only proceed if a dataset exists for the scenario's area code.
+      next unless Atlas::Dataset.exists?(scenario.area_code)
+
+      dataset = Atlas::Dataset.find(scenario.area_code)
+
+      # Rename the user values based on the new key mapping.
+      rename_keys(scenario, NEW_SHARE_MAPPING)
+
+      # Check if the scenario has any attached curves (user profiles).
+      if USER_PROFILE_KEYS.any? { |profile| scenario.attachment?(profile) }
+        # Fetch the current setup of attached curves or dataset profiles.
+        current_setup = fetch_attached_curves(scenario, dataset)
+        # Create and set up a custom profile based on the current setup.
+        make_custom_profile(scenario, current_setup)
+      end
+
+      # Save the scenario and track the migration.
+      scenario.save
+      track_scenario_migration(scenario)
+    end
+  end
+
+  private
+
+  # Sets up a custom profile with a default share distribution and assigns it to profile 5.
+  def setup_custom_profile(scenario, custom_profile)
+    # Reset all new profile shares to zero.
+    NEW_SHARE_KEYS.each { |key| scenario.user_values[key] = 0 }
+
+    # Detach any existing electric vehicle profile curve attachments.
+    USER_PROFILE_KEYS.each do |profile_key|
+      attachment = scenario.attachment(profile_key)
+      CurveHandler::DetachService.call(attachment) if attachment
+    end
+
+    # Set the custom profile share to 100% for 'electric_vehicle_profile_5_curve'.
+    scenario.user_values['transport_car_using_electricity_custom_profile_charging_share'] = 100.0
+    attach_custom_profile(scenario, custom_profile)
+  end
+
+  # Creates a custom profile based on the current profiles and their shares.
+  def make_custom_profile(scenario, current_setup)
+    mix = {}
+
+    # Prepare a mix of profiles and their corresponding shares.
+    NEW_SHARE_KEYS.each_with_index do |key, index|
+      share = scenario.user_values[key] || 0.0
+      profile_key = USER_PROFILE_KEYS[index]
+      profile = current_setup[profile_key]
+
+      mix[profile] = share if profile && share.positive?
+    end
+
+    # Exit if no profiles were found in the mix.
+    return if mix.empty?
+
+    # Aggregate the profiles and shares to create a custom profile.
+    custom_profile = curve_aggregator(mix)
+    return if custom_profile.nil?
+
+    # Set up the custom profile in the scenario.
+    setup_custom_profile(scenario, custom_profile)
+  end
+
+  # Renames keys in the scenario's user values based on the provided mapping.
+  def rename_keys(scenario, new_mapping)
+    new_mapping.each do |old_key, new_key|
+      # Transfer values from old keys to new keys, defaulting to 0 if the old key is not present.
+      scenario.user_values[new_key] = scenario.user_values.delete(old_key) || 0.0
```

Comment:
```suggestion
      # Transfer values from old keys to new keys, if the old key is present.
      next unless scenario.user_values.key? (old_key)

      scenario.user_values[new_key] = scenario.user_values.delete(old_key)
```

If none of the sliders in the group was touched, and now they all default to 0.0, the group does not sum too 100% and the scenario will break. It's best to just skip when the key was not there.